### PR TITLE
[acceptance-test] Bind two backend services by creating 1 SBR to a single application

### PR DESCRIPTION
**Pull request by pratikjagrut**
### Motivation

https://issues.redhat.com/browse/APPSVC-660

### Testing

`make test-acceptance VERBOSE=3`

For further more details refer the CONTRIBUTING.md

### CODE REVIEW

File: test/acceptance/features/bindAppToMultipleServices.feature [hunk 1]
```diff
@@ -55,3 +55,56 @@ Feature: Bind a single application to multiple services
         And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding "binding-request-2" should be changed to "True"
         And "nodejs-app" deployment must contain SBR name "binding-request-1"
         And "nodejs-app" deployment must contain SBR name "binding-request-2"
+
+        Scenario: Bind two backend services by creating 1 SBR to a single application
```
Comment (Avni-Sharma) [inline test/acceptance/features/bindAppToMultipleServices.feature:59]: for maintaining consistency we can say `two backend services by creating one SBR` or ` 2 backend services by creating 1 SBR` . Former sounds better as we are maintaining it in our feature files.

Comment (Avni-Sharma) [reply test/acceptance/features/bindAppToMultipleServices.feature:59]: Also in our tests, we are mentioning binding an app to a service, instead of service to an app

Comment (pratikjagrut) [reply test/acceptance/features/bindAppToMultipleServices.feature:59]: > for maintaining consistency we can say `two backend services by creating one SBR` or ` 2 backend services by creating 1 SBR` . Former sounds better as we are maintaining it in our feature files.

This name is consistent with the scenario already present in the file. https://github.com/redhat-developer/service-binding-operator/blob/master/test/acceptance/features/bindAppToMultipleServices.feature#L11

File: test/acceptance/features/bindAppToMultipleServices.feature [hunk 2]
```diff
@@ -55,3 +55,56 @@ Feature: Bind a single application to multiple services
...
+        Given Imported Nodejs application "nodejs-app-2-services-1-sbr" is running
```
Comment (pedjak) [inline test/acceptance/features/bindAppToMultipleServices.feature:60]: does this application really uses two databases?

File: test/acceptance/features/bindAppToMultipleServices.feature [hunk 3]
```diff
@@ -55,3 +55,56 @@ Feature: Bind a single application to multiple services
...
+                    name: internal-db
+                -   group: postgresql.baiju.dev
+                    version: v1alpha1
+                    kind: Database
+                    name: external-db
+            """
+
+        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding "binding-request-2-services-1-sbr" should be changed to "True"
+        And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding "binding-request-2-services-1-sbr" should be changed to "True"
+        And "nodejs-app-2-services-1-sbr" deployment must contain SBR name "binding-request-2-services-1-sbr"
```
Comment (pedjak) [inline test/acceptance/features/bindAppToMultipleServices.feature:108]: we do not check if app is really running correctly.

File: test/acceptance/features/bindAppToMultipleServices.feature (hunk 2 above)
Comment (pratikjagrut) [reply test/acceptance/features/bindAppToMultipleServices.feature:60]: NO, the app is not designed that way.

File: test/acceptance/features/bindAppToMultipleServices.feature (hunk 3 above)
Comment (pratikjagrut) [reply test/acceptance/features/bindAppToMultipleServices.feature:108]: We're checking if the deployment has the env var or not.

File: test/acceptance/features/bindAppToMultipleServices.feature (hunk 1 above)
Comment (pedjak) [reply test/acceptance/features/bindAppToMultipleServices.feature:59]: @pratikjagrut - the scenario description look good.

@Avni-Sharma - `two backend services by creating one SBR` or `2 backend services by creating 1 SBR` misses a verb, i.e. action.

File: test/acceptance/features/bindAppToMultipleServices.feature (hunk 3 above)
Comment (pedjak) [reply test/acceptance/features/bindAppToMultipleServices.feature:108]: we are checking only the binding secret is referred in the deployment, but not if env var is really available within app process.

File: test/acceptance/features/bindAppToMultipleServices.feature [hunk 4]
```diff
@@ -6,12 +6,32 @@ Feature: Bind a single application to multiple services
...
         Given Namespace [TEST_NAMESPACE] is used
         * Service Binding Operator is running
-        * PostgreSQL DB operator is installed
+        * OLM Operator "backend" is running
 
     Scenario: Bind two backend services by creating 2 SBRs to a single application
-        Given Imported Nodejs application "nodejs-app" is running
-        * DB "db-demo-1" is running
-        * DB "db-demo-2" is running
+        Given Generic test application "myapp-2sbrs" is running
```
Comment (pedjak) [inline test/acceptance/features/bindAppToMultipleServices.feature:12]: Focus of this PR (according to the summary) is about creating a test for binding two services with single SB. Hence, this scenario should not be touched here. Please revert the changes, and apply the in another focused PR.

File: test/acceptance/features/bindAppToMultipleServices.feature [hunk 5]
```diff
@@ -38,20 +58,79 @@ Feature: Bind a single application to multiple services
...
+                    version: v1
+                    kind: Backend
+                    name: external-db-1sbr
+            """
+
+        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding "binding-request-1sbr" should be changed to "True"
+        And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding "binding-request-1sbr" should be changed to "True"
+        And Secret "binding-request-1sbr" contains "BACKEND_HOST_INTERNAL_DB" key with value "internal.db.stable.example.com"
+        And Secret "binding-request-1sbr" contains "BACKEND_HOST_EXTERNAL_DB" key with value "external.db.stable.example.com"
+        And "myapp-1sbr" deployment must contain SBR name "binding-request-1sbr"
```
Comment (pedjak) [inline test/acceptance/features/bindAppToMultipleServices.feature:134]: No need to check this, because the last steps check that implicitly. Also, injecting env vars via secrets is an implementation details, and user must not be aware of that. What she/he really cares is about what env vars are visible to the application.

**Review by pedjak — CHANGES_REQUESTED** [review]

File: test/acceptance/features/bindAppToMultipleServices.feature (hunk 4 above)
Comment (pratikjagrut) [reply test/acceptance/features/bindAppToMultipleServices.feature]: Changes reverted.

File: test/acceptance/features/bindAppToMultipleServices.feature (hunk 5 above)
Comment (pratikjagrut) [reply test/acceptance/features/bindAppToMultipleServices.feature]: Done.

**Comment by pratikjagrut** [comment]
Forced push for squashing.

**Review by pedjak — APPROVED** [review]
